Makefile.toml

```diff
@@ -201,11 +201,17 @@ script = [
 '''
 export VARIANT="${BUILDSYS_VARIANT}"
 
+# Exclude Kubernetes specific packages from being built if the variant isn't Kubernetes specific
+if ! echo "${VARIANT}" | grep -q "k8s"; then
+    EXCLUDE="--exclude static-pods pluto"
+fi
+
```

Comment:
There has to be a conditional somewhere for this unit test task to work. Either we continue conditionally compile variant specific binaries, or we do this.

Reply:
Yeah, I realize that's the problem we're solving for. With ecs-settings-applier, the build.rs exports a cfg and the 'guts' are not compiled unless its ecs.

The problem with this approach is that the rust workspace is no longer usable on its own and we're moving toward a reliance on the makefile for things to work at all.

Both solutions are undesirable, so I'm not demanding a change here, but inviting discussion.

Reply:
Sorry for jumping in between the conversation.
```
cd sources
export VARIANT=aws-ecs-1 && cargo build
```
@webern  Why won't this work ? This conditional is not inside `sources` dir packages right ? Not able to figure out how it affects.

Reply:
I'm assuming that the reason @etungsten added this to the makefile is because the `pluto` and `static-pods` packages will fail to compile unless the `model` package is set to use `k8s` settings. If we try to compile the workspace for ECS, the `model` will have the wrongs structs and our build will fail.

Reply:
Specifically `tasks.unit-tests` will fail because it just compiles everything in the workspace with the same VARIANT env var.

Regarding the `cd sources; export VARIANT=aws-ecs-1 && cargo build` use case. I feel like that's somewhat unusual. What reasons are there for people wanting to build the entire workspace without going through the Makefile? I would understand if people wanted to build specific workspace components for testing or w.e. But if they want to build everything in the workspace for use. shouldn't they just go through building the OS package via the Makefile?

Reply:
I'm OK with your original conditional compilation for static-pods if it makes the overall workspace easier to build. I like the property that these crates can be built without bringing the entire build system to bear.

I'd prefer to get to the point where the entire sources tree can be built for each model simultaneously, and although the `Makefile.toml` workaround doesn't preclude that, it may invite us to continue segmenting the tree in ad-hoc ways rather than driving toward that goal.

Reply:
I use the workspace without bringing the build system to bear all the time. I'm not sure how to describe the use case, but having a workspace that can't be compiles with `cargo build` or `cargo test` seems to defeat the purpose of a workspace. Right now the only "gotchya" is that the `VARIANT` env is needed, and the error messages document this very well (I often forget, read the error then export a VARIANT).

I think the options are:

1) Merge this as is with the  `Makefile.toml` changes, and no longer can we `cargo build` the workspace (without adding the `--exclude` statements depending on `VARIANT`, or
2) Change to the `cfg` solution that we used with `ecs-settings-applier` so that we can always build all of the workspace, but the incompatible code is 'commented out'.

@bcressey are you saying that you choose `#2` and that, because the workspace no longer builds, we may be more motivated to fix it?

Reply:
I'll revert back to conditionally compiling static-pods since that's what people seem to prefer so far.

Reply:
No, I was agreeing with you and I prefer `#2`.